xgcm/test/test_metrics.py

```diff
@@ -179,6 +179,16 @@ def test_get_metric_orig(
         assert metric.equals(expected)
 
 
+def test_get_metric_with_no_matching_axis():
+    # the metric axis should be present in array
+    ds, coords, metrics = datasets_grid_metric("C")
+    grid = Grid(ds, coords=coords, metrics=metrics)
+
+    match_message = "Did not find single matching dimension"
```

Comment:
Isn't this the same logic as in `test_metric_axes_missing_from_array`?

Reply:
Yep, I wrote this initially to test including the line `self._get_dims_from_axis(array, frozenset(axes))` in get_metric. I'll take it out, because `test_metric_axes_missing_from_array` does the same thing.